### PR TITLE
fix: throttle failed JWT profile refresh attempts

### PR DESCRIPTION
**Pull request by thienchi2109**
## Summary
- Add `lastRefreshAttemptAt` to JWT tokens so failed/no-profile refresh attempts participate in cooldown.
- Keep `lastRefreshAt` reserved for successful profile refreshes.
- Cover failed RPC and no-profile retry throttling in focused JWT cooldown tests.

Fixes #431

## Verification
- RED: `node scripts/npm-run.js run test:run -- src/auth/__tests__/auth-config.jwt-cooldown.test.ts` failed 2 expected tests before implementation.
- `node scripts/npm-run.js run verify:no-explicit-any`
- `node scripts/npm-run.js run typecheck`
- `node scripts/npm-run.js run test:run -- src/auth/__tests__/auth-config.jwt-cooldown.test.ts src/auth/__tests__/auth-config.jwt-rpc.test.ts`
- `node scripts/npm-run.js npx react-doctor@latest . --verbose -y --project nextn --offline --diff main`
- `git diff --check`

---
## Summary by cubic
Throttle JWT profile refresh retries when the profile fetch fails or returns no profile by tracking a new `lastRefreshAttemptAt` timestamp; successful refreshes continue to use `lastRefreshAt`. Improves cooldown telemetry and fixes #431.

- **Bug Fixes**
  - Enforce `PROFILE_REFRESH_INTERVAL_MS` using the latest of `lastRefreshAt` and `lastRefreshAttemptAt`, unless `trigger === "update"`.
  - Set `lastRefreshAttemptAt` before calling `get_session_profile_for_jwt` to prevent rapid retry loops on RPC failure or no profile.
  - Emit accurate telemetry for cooldown skips (compute `hasLastRefreshAt` from `lastRefreshAt` only) and add tests for failed/no-profile retries and telemetry.

<sup>Written for commit 44d29985e5c4a17cd6d8319cdf77cd1861909fe7. Summary will update on new commits.</sup>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved JWT refresh-cooldown mechanism with enhanced retry throttling for profile refresh attempts.
  * Better handling of profile refresh failures and edge cases with separate attempt tracking.

* **Tests**
  * Added comprehensive test coverage verifying retry throttling behavior when profile refresh fails or returns no data.

[Review Change Stack](https://app.coderabbit.ai/change-stack/thienchi2109/qltbyt-nam-phong/pull/481)

### CODE REVIEW

File: src/auth/__tests__/auth-config.jwt-cooldown.test.ts [hunk 1]
```diff
@@ -136,6 +136,10 @@ async function runJwt(args: Partial<JwtArgs> & Pick<JwtArgs, "token">) {
   } as JwtArgs)
 }
 
+function profileRefreshRpcCalls() {
+  return supabaseState.rpcCalls.filter((call) => call.fn === "get_session_profile_for_jwt")
+}
```
[the remark this answers is not in the document]
Comment (thienchi2109) [reply src/auth/__tests__/auth-config.jwt-cooldown.test.ts]: Fixed in 44d29985: `profileRefreshRpcCalls` now declares an explicit return type while preserving the existing filter behavior. Re-ran `verify:no-explicit-any`, `typecheck`, focused auth JWT tests, React Doctor diff, and `git diff --check`.
